From 9d42a89b917bd7d1b3c9af602cbb6bc36ab4196c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:05:42 -0400
Subject: [tools] perf/session: Change header.misc dump from decimal to hex

Message-id: <1410181752-25631-98-git-send-email-jolsa@redhat.com>
Patchwork-id: 91769
O-Subject: [PATCH RHEL7.1 BZ1134356 097/307] perf session: Change header.misc dump from decimal to hex
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 0ea590ae8198547d5898c72b04fa9d8f23bd0b8f
Author: Don Zickus <dzickus@redhat.com>
Date: Tue Feb 25 22:43:46 2014 -0500

description
===========
When printing the raw dump of a data file, the header.misc is
printed as a decimal.  Unfortunately, that field is a bit mask, so
it is hard to interpret as a decimal.

Print in hex, so the user can easily see what bits are set and more
importantly what type of info it is conveying.

V2: add 0x in front per Jiri Olsa
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/session.c b/tools/perf/util/session.c
index 1d555d6..55960f2 100644
--- a/tools/perf/util/session.c
+++ b/tools/perf/util/session.c
@@ -794,7 +794,7 @@ static void dump_sample(struct perf_evsel *evsel, union perf_event *event,
  if (!dump_trace)
   return;
 
- printf("(IP, %d): %d/%d: %#" PRIx64 " period: %" PRIu64 " addr: %#" PRIx64 "\n",
+ printf("(IP, 0x%x): %d/%d: %#" PRIx64 " period: %" PRIu64 " addr: %#" PRIx64 "\n",
         event->header.misc, sample->pid, sample->tid, sample->ip,
         sample->period, sample->addr);
 
-- 
1.7.1